File: AIResources/Python/ExampleBot.py
from hlt import *
from networking import *

sock = connectToGame()
playerTag, gameMap = getInit(sock)
sendInit(sock)

while True:
	moves = []
	sendMessages = [Message(ATTACK, playerTag, 1 if playerTag == 2 else 2, playerTag)]
	gameMap, recievedMessages = getFrame(sock)
	for message in recievedMessages:
		print("Message: ")
		print(message.type)
		print(message.senderID)
		print(message.recipientID)
		print(message.targetID)
	for y in range(0, len(gameMap.contents)):
		for x in range(0, len(gameMap.contents[y])):
			site = gameMap.contents[y][x]
			if site.owner == playerTag:
				moves.append(Move(Location(x, y), int(random.random() * 4)))
	sendFrame(sock, moves, sendMessages)

File: AIResources/Java/ExampleBot.java
import java.net.*;
import java.util.ArrayList;

public class ExampleBot
{
    public static void main(String[] args) {
        System.out.println("\u000c");
        Socket sock = Networking.connectToGame();
        
        InitPackage iPackage = Networking.getInit(sock);
        short playerTag = iPackage.playerTag;
        Map gameMap = iPackage.map;

        Networking.sendInit(sock);

        while(true) {
            ArrayList<Move> moves = new ArrayList<Move>();
            ArrayList<Message> sendMessages = new ArrayList<Message>();
            
            sendMessages.add(new Message(MessageType.STOP_ATTACK, playerTag, playerTag != 1 ? 1 : 2, playerTag));
            
            FramePackage fPackage = Networking.getFrame(sock);
            gameMap = fPackage.map;
            ArrayList<Message> recievedMessages = fPackage.messages;
            
            for(Message message : recievedMessages) {
                System.out.println("Message: " + message.type.getValue() + " " + message.senderID + " " + message.recipientID + " " + message.targetID);
            }
            
            for(int y = 0; y < gameMap.contents.size(); y++) {
                for(int x = 0; x < gameMap.contents.get(y).size(); x++) {
                    Site site = gameMap.contents.get(y).get(x);
                    if(site.owner == playerTag) {
                        byte dir = Direction.randomDirection();
                        moves.add(new Move(new Location((short)x, (short)y), dir));
                    }
                }
            }
            
            Networking.sendFrame(sock, moves, sendMessages);
        }
    }
}
